### ISSUE TITLE
Missing type parameter in impl Default suggestion.

### ISSUE THREAD

**Issue by cassiersg** [rust-lang/rust-clippy#6933]
When a type with a type parameter as a method `pub fn new() -> Self`, clippy suggests to implement `Default`. However the suggested implementation lacks the type parameter.

For instance, when running `cargo clippy` on

```rust
#![allow(dead_code)]
pub struct Foo<T>(std::marker::PhantomData<T>);

impl<T> Foo<T> {
    pub fn new() -> Self {
        Self(Default::default())
    }
}

pub struct Bar<T>(std::marker::PhantomData<T>);

impl<T: Copy> Bar<T> {
    pub fn new() -> Self {
        Self(Default::default())
    }
}
```
the output is
```
warning: you should consider adding a `Default` implementation for `Foo<T>`
 --> src/lib.rs:5:5
  |
5 | /     pub fn new() -> Self {
6 | |         Self(Default::default())
7 | |     }
  | |_____^
  |
  = note: `#[warn(clippy::new_without_default)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default
help: try this
  |
4 | impl Default for Foo<T> {
5 |     fn default() -> Self {
6 |         Self::new()
7 |     }
8 | }
  |

warning: you should consider adding a `Default` implementation for `Bar<T>`
  --> src/lib.rs:13:5
   |
13 | /     pub fn new() -> Self {
14 | |         Self(Default::default())
15 | |     }
   | |_____^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default
help: try this
   |
12 | impl Default for Bar<T> {
13 |     fn default() -> Self {
14 |         Self::new()
15 |     }
16 | }
   |
```
while the impl suggestions should be
```rust
impl<T> Default for Foo<T> {
...
}
impl<T: Copy> Default for Bar<T> {
...
}
```
(notice the `: Copy` bound).

### Meta

- `cargo clippy -V`: clippy 0.0.212 (cb75ad5 2021-02-10)
- `rustc -Vv`:
```
 rustc 1.50.0 (cb75ad5db 2021-02-10)
binary: rustc
commit-hash: cb75ad5db02783e8b0222fee363c5f63f7e2cf5b
commit-date: 2021-02-10
host: x86_64-unknown-linux-gnu
release: 1.50.0
```
